src/View/Helper/CrudViewHelper.php

```diff
@@ -326,7 +335,7 @@ public function createRelationLink($alias, $relation, $options = [])
                 'action' => 'add',
                 '?' => [
                     $relation['foreignKey'] => $this->getViewVar('primaryKeyValue'),
-                    '_redirect_url' => $this->request->here
+                    '_redirect_url' => $this->request->getUri()->getPath()
```

Comment:
I am not sure if this returns same path as request->here() which isn't deprecated.